text/0000-api-version-compat.md

```diff
@@ -0,0 +1,121 @@
+# Meta
+[meta]: #meta
+- Name: Lifecycle API version support changes
+- Start Date: 2020-05-14
+- Author(s): [Emily Casey](https://github.com/ekcasey)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+
+# Summary
+[summary]: #summary
+
+To enable smoother migrations and upgrades this RFC proposes the following changes to the API version compatibility rules:
+
+1. When the lifecycle increments a supported API in its descriptor file to `0.x`, this implies support for `0.2-0.x` versions of that API
+1. The spec may make changes it wishes in a new 0.x API so long as those changes do not preclude buildpacks implementing different 0.x APIs from running together (e.g changes to the contractual build plan that cannot be bridged by the lifecycle)
+1. The platform can configure which version of the API the lifecycle implements
+1. When interacting with a buildpack, the lifecycle will use the API declared in each buildpack's `buildpack.toml`
+
+The lifecycle will assume the burden of maintaining compatibility, reducing the burden on platform and buildpack authors.  
+
+# Motivation
+[motivation]: #motivation
+
+Because there are far fewer lifecycle implementation than platforms or buildpacks, the lifecycle should assume the burden of maintaining compatibility. This will free platform authors to consume new lifecycles without worry and free buildpack authors to implement new APIs at their own pace. 
```

Comment:
> buildpack authors to implement new APIs at their own pace

With the caveat that they must implement all or none of the changes for a given API version because there is no concept of deprecation, right?

Reply:
Added deprecations back.